### PR TITLE
More fixups

### PR DESCRIPTION
**Pull request by msukkar**
@aw3as want to take a look? Some more minor fixes on the way.

### CODE REVIEW

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -254,20 +254,115 @@ select {
...
+  // set height to length of navbar and width 
+  // to bootstrap default
   border: none;
-  height: $navbar-height;;
-  margin-top: 0px;
-  margin-bottom: 0px;
-  .navbar-default &:hover, .navbar-default &:focus{
-    background-color: inherit;
-    .icon-bar {
+  height: 60px;
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: can/are we using variables for these numbers? heigh, width,etc

File: app/assets/stylesheets/application.scss [hunk 2]
```diff
@@ -254,20 +254,115 @@ select {
     }
 }
 
-.navbar-toggle {
+
+// code below handles transition navbar button transition from hamburger to x
+$transition-dur: .2s;
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: move these to variables?

File: app/assets/stylesheets/application.scss [hunk 3]
```diff
@@ -254,20 +254,115 @@ select {
...
+// same as native bootstrap button-size
+$button-size: 22px;
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: move these to variables?

File: app/assets/stylesheets/sidebar.scss [hunk 4]
```diff
@@ -65,9 +65,8 @@
         text-align: center;
         position: relative;
         float: right;
-        margin-right: 25%;
         margin-top: 6px;
-        margin-left: auto;
+        margin-right: 50px;
```
Comment (aw3as) [inline app/assets/stylesheets/sidebar.scss]: assuming this for the beta-tag, can you check different browsers and screen sizes?

File: app/assets/stylesheets/welcome.scss [hunk 5]
```diff
@@ -6,6 +6,7 @@
 
 #panel-login {
   width: 100%;
+  margin-left: 0px !important;
```
Comment (aw3as) [inline app/assets/stylesheets/welcome.scss]: is !important necessary here

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -61,6 +90,9 @@ var ready = function() {
 	$(document).keydown(function(e) {
 		if (e.which === 27 && $(window).width() < 850) {
 			$("aside").hide('slide', toggleSpeed);
+			if (open == true){
```
Comment (aw3as) [inline app/assets/javascripts/application.js]: please can we leave this in for posterity

File: app/assets/stylesheets/application.scss [hunk 7]
```diff
@@ -254,20 +254,115 @@ select {
...
+
+.lines {
+  @include line;
+  position: relative; 
+
+  // inserts top and bottom lines via css. Are not actually in DOM
+  &:before, &:after {
+   @include line;
+    position: absolute;
+    left:0;
```
Comment (aw3as) [inline app/assets/stylesheets/application.scss]: can you format this <3

File: app/assets/javascripts/application.js (hunk 6 above)
Comment (msukkar) [inline app/assets/javascripts/application.js]: lol no

Comment (keenadams) [inline app/assets/javascripts/application.js]: lol this is officially called a nipun

**Comment by msukkar** [comment]
@aw3as Want to take another look? Refactored this. I checked that nothing broke but you may want to run locally to double check me.

**Comment by msukkar** [comment]
@keenadams want to review this?

File: app/assets/javascripts/application.js [hunk 8]
```diff
@@ -34,62 +34,13 @@
...
-		if ($("#search-query").val() == "") {
-			$(".nav-row").each(function() {
-				$(this).slideDown();
-			});
-
-			$(".submit-row").slideUp();
-		}
-	});
-
 	$("#word-cloud-switch").bootstrapSwitch({
```
Comment (aw3as) [inline app/assets/javascripts/application.js]: does this functionality still exist?

File: app/assets/javascripts/sidebar.js [hunk 9]
```diff
@@ -0,0 +1,160 @@
...
+		if ($("#search-query").val() == "") {
+			$(".nav-row").each(function() {
+				$(this).slideDown();
+			});
+
+			$(".submit-row").slideUp();
+		}
+	});
+
+	$('.professor_link').bind('ajax:success', function(xhr, data, status) {
```
Comment (aw3as) [inline app/assets/javascripts/sidebar.js]: what's this for?

File: app/assets/javascripts/sidebar.js [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+		},
+		minLength: 2,
+		select: function(event, ui) {
+			window.location = "/courses/" + ui.item.course_id + '?p=all';
+			return false;
+		}
+	});
+
+
+	var prof_ajax = $.ajax();
```
Comment (aw3as) [inline app/assets/javascripts/sidebar.js]: ???

File: app/assets/javascripts/sidebar.js [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+							text: this.last_name + ", " + this.first_name
+						}));
+						$('#prof_list').append($("<br/>", {}));
+					}
+				});
+			}
+		});
+	});
+
+	jQuery.ajaxSetup({
```
Comment (aw3as) [inline app/assets/javascripts/sidebar.js]: is this a global ajax override?

File: app/assets/stylesheets/header.scss [hunk 12]
```diff
@@ -54,4 +56,125 @@
...
+}
+
+// code below handles transition navbar button transition from hamburger to x
+$transition-dur: .2s;
+// same as native bootstrap button-width
+$button-width: 22px;
+
+// line element. Mixin so that it can be reused
+@mixin line {
+  display: inline-block;
```
Comment (aw3as) [inline app/assets/stylesheets/header.scss]: fix inconsistent indenting

File: app/assets/javascripts/application.js (hunk 8 above)
Comment (msukkar) [inline app/assets/javascripts/application.js]: see below

File: app/assets/javascripts/sidebar.js (hunk 9 above)
Comment (msukkar) [inline app/assets/javascripts/sidebar.js]: I believe this handles sliding sidebar rows back down when losing focus. Moved from application.js

File: app/assets/stylesheets/header.scss (hunk 12 above)
Comment (msukkar) [inline app/assets/stylesheets/header.scss]: done

File: app/assets/javascripts/sidebar.js (hunk 11 above)
Comment (msukkar) [inline app/assets/javascripts/sidebar.js]: nah. moved from application.js

File: app/assets/javascripts/application.js (hunk 8 above)
Comment (msukkar) [inline app/assets/javascripts/application.js]: Alan actually meant the word cloud switch. That and the doge functionality have now been removed.

**Pull request merged**